src/lib/server/tools/index.ts

```diff
@@ -0,0 +1,37 @@
+import type { Assistant } from "$lib/types/Assistant";
+import type { Conversation } from "$lib/types/Conversation";
+import type { Message } from "$lib/types/Message";
+import type { MessageUpdate } from "$lib/types/MessageUpdate";
+import type { Tool, ToolResult } from "$lib/types/Tool";
+
+import calculator from "./calculator";
+import directlyAnswer from "./directlyAnswer";
+import imageEditing from "./images/editing";
+import imageGeneration from "./images/generation";
+import parsePdf from "./parse/pdf";
+import fetchUrl from "./web/url";
+import websearch from "./web/search";
+
+interface BackendToolContext {
+	conv: Conversation;
```

Comment:
why is entire `Conversation` rather than `Conversation.Id` is preferred here?

Reply:
We need a few properties off the conversation object such as the `assistantId` and that information has already been grabbed in the `conversation/[id]/+server.ts`